### ISSUE TITLE
Error in MIME::Types::WarnLogger

### ISSUE THREAD

**Issue by milushov** [mime-types/ruby-mime-types#175]
I installed the `rest-client` gem, and the project started failing due to some error in the `mime-types` gem.

Gemfile:
```
...
gem "rest-client", '~> 1.8'
```

Gemfile.lock:
```
...
mime-types (2.99.3)
...
```

Ruby version:
```
ARG RUBY_VERSION=3.2.2
FROM ruby:${RUBY_VERSION}-slim
```

Rails version:
```
7.1.0.alpha
```

I see the error, is there any ideas how to fix this?

```
╰─$ make rc
docker-compose run --rm app rails console
[+] Running 2/0
 ⠿ Container awesome-pg-1     Running                                                                                                                                          0.0s
 ⠿ Container awesome-redis-1  Running                                                                                                                                          0.0s
DEPRECATION WARNING: DeprecatedConstantAccessor.deprecate_constant without a deprecator is deprecated (called from <module:Authenticatable> at /bundle/ruby/3.2.0/gems/devise-4.9.2/lib/devise/models/authenticatable.rb:65)
<internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require': --> /bundle/ruby/3.2.0/gems/mime-types-2.99.3/lib/mime/types/logger.rb
_1 is reserved for numbered parameter_2 is reserved for numbered parameter_3 is reserved for numbered parameter
   5  module MIME
   6    class Types
  13      class WarnLogger < ::Logger #:nodoc:
> 26        def initialize(_1, _2 = nil, _3 = nil)
> 30        end
  31      end
  34    end
  35  end
/bundle/ruby/3.2.0/gems/mime-types-2.99.3/lib/mime/types/logger.rb:26: _1 is reserved for numbered parameter (SyntaxError)
/bundle/ruby/3.2.0/gems/mime-types-2.99.3/lib/mime/types/logger.rb:26: _2 is reserved for numbered parameter
/bundle/ruby/3.2.0/gems/mime-types-2.99.3/lib/mime/types/logger.rb:26: _3 is reserved for numbered parameter
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /bundle/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /bundle/ruby/3.2.0/gems/zeitwerk-2.6.8/lib/zeitwerk/kernel.rb:38:in `require'
	from /bundle/ruby/3.2.0/gems/mime-types-2.99.3/lib/mime/types/deprecations.rb:3:in `<main>'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /bundle/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /bundle/ruby/3.2.0/gems/zeitwerk-2.6.8/lib/zeitwerk/kernel.rb:38:in `require'
	from /bundle/ruby/3.2.0/gems/mime-types-2.99.3/lib/mime/types.rb:3:in `<main>'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /bundle/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /bundle/ruby/3.2.0/gems/zeitwerk-2.6.8/lib/zeitwerk/kernel.rb:38:in `require'
	from /bundle/ruby/3.2.0/gems/rest-client-1.8.0/lib/restclient/request.rb:2:in `<main>'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /bundle/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /bundle/ruby/3.2.0/gems/zeitwerk-2.6.8/lib/zeitwerk/kernel.rb:38:in `require'
	from /bundle/ruby/3.2.0/gems/rest-client-1.8.0/lib/restclient.rb:10:in `<main>'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /bundle/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /bundle/ruby/3.2.0/gems/zeitwerk-2.6.8/lib/zeitwerk/kernel.rb:38:in `require'
	from /bundle/ruby/3.2.0/gems/rest-client-1.8.0/lib/rest-client.rb:2:in `<main>'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /bundle/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /bundle/ruby/3.2.0/gems/zeitwerk-2.6.8/lib/zeitwerk/kernel.rb:38:in `require'
	from /bundle/gems/bundler-2.4.15/lib/bundler/runtime.rb:60:in `block (2 levels) in require'
	from /bundle/gems/bundler-2.4.15/lib/bundler/runtime.rb:55:in `each'
	from /bundle/gems/bundler-2.4.15/lib/bundler/runtime.rb:55:in `block in require'
	from /bundle/gems/bundler-2.4.15/lib/bundler/runtime.rb:44:in `each'
	from /bundle/gems/bundler-2.4.15/lib/bundler/runtime.rb:44:in `require'
	from /bundle/gems/bundler-2.4.15/lib/bundler.rb:187:in `require'
	from /app/config/application.rb:22:in `<main>'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /bundle/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /bundle/ruby/3.2.0/bundler/gems/rails-c9cdd800d30e/railties/lib/rails/command/actions.rb:15:in `require_application!'
	from /bundle/ruby/3.2.0/bundler/gems/rails-c9cdd800d30e/railties/lib/rails/command/environment_argument.rb:31:in `require_application!'
	from /bundle/ruby/3.2.0/bundler/gems/rails-c9cdd800d30e/railties/lib/rails/command/actions.rb:19:in `boot_application!'
	from /bundle/ruby/3.2.0/bundler/gems/rails-c9cdd800d30e/railties/lib/rails/commands/console/console_command.rb:101:in `perform'
	from /bundle/ruby/3.2.0/gems/thor-1.2.2/lib/thor/command.rb:27:in `run'
	from /bundle/ruby/3.2.0/gems/thor-1.2.2/lib/thor/invocation.rb:127:in `invoke_command'
	from /bundle/ruby/3.2.0/bundler/gems/rails-c9cdd800d30e/railties/lib/rails/command/base.rb:179:in `invoke_command'
	from /bundle/ruby/3.2.0/gems/thor-1.2.2/lib/thor.rb:392:in `dispatch'
	from /bundle/ruby/3.2.0/bundler/gems/rails-c9cdd800d30e/railties/lib/rails/command/base.rb:74:in `perform'
	from /bundle/ruby/3.2.0/bundler/gems/rails-c9cdd800d30e/railties/lib/rails/command.rb:71:in `block in invoke'
	from /bundle/ruby/3.2.0/bundler/gems/rails-c9cdd800d30e/railties/lib/rails/command.rb:149:in `with_argv'
	from /bundle/ruby/3.2.0/bundler/gems/rails-c9cdd800d30e/railties/lib/rails/command.rb:69:in `invoke'
	from /bundle/ruby/3.2.0/bundler/gems/rails-c9cdd800d30e/railties/lib/rails/commands.rb:18:in `<main>'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /bundle/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /app/bin/rails:4:in `<main>'
make: *** [rc] Error 1
```

**Comment by milushov** [comment]
Somehow I have old code in the log:

```
   5  module MIME
   6    class Types
  13      class WarnLogger < ::Logger #:nodoc:
> 26        def initialize(_1, _2 = nil, _3 = nil)
> 30        end
  31      end
  34    end
  35  end
  ```

But it was changed 4y ago: https://github.com/mime-types/ruby-mime-types/commit/c44673179d24e495e5fb93282a87d37f09925d25#diff-3c02420bac1fd3f383f605fd2f8a763ae1aa58cd09cb530612ce8adf3045f30c

How it's possible?

**Comment by milushov** [comment]
Owwhhh rest-client is sooo outdated omg, make sense now

**Comment by halostatue** [comment]
Yeah. I closed rest-client/rest-client#445 in September 2022 after seven years of no response to a trivial PR. It is a project best avoided at this point.
